### DISCUSSION TITLE
sqlalchemy-2.0.0 fails to upsert data on a mysql 5.7 server

### DISCUSSION THREAD

**Discussion by valdrinsherifi** [sqlalchemy/sqlalchemy#9216] [Usage Questions]
While trying to collect the dependencies on a project using venv and pip freeze, I was installing the latest package versions.
So I was using sqlalchemy version 2.0.0 in a venv and upserting data failed what previously working with version 1.4.44.

I was doing the something similar to:

```
def upsert():
    with engine.connect() as db:
        try:
            db.execute(
                text(UPSERT_QUERY)
                    )
        except Exception as e:
            print(e)
```

Please keep in mind that there is no issue with the engine, as other operations are working just fine. Also it is failing to catch any exceptions. Also I used the raw query in datagrip and it successfully executed.

I just wanted to report this what could perhaps be a bug.
I just reverted to version 1.4.44 and everything is working fine.
BTW I am using python 3.8.6

**Comment by CaselIT** [comment]
Hi,

you are not committing the upsert, add a `db.commit()` after the execution or use instead `engine.begin()`.

If this worked on 1.4 then you probably have autocommit setup in a way that's no longer supported in 2.0. Look at the migration: https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#library-level-but-not-driver-level-autocommit-removed-from-both-core-and-orm
